R/rtf_static_forestly.R

```diff
@@ -0,0 +1,213 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Output static forest plot
+#'
+#' @param outdata An `outdata` object created by [format_ae_forestly()].
+#' @param plot_calls A list or vector of function calls.
+#' @param source A character value of the data source.
+#' @param parameter A character value of parameter term name.
+#' @param n_rows An integer value of the number of rows per page in a plot.
+#' @param fig_size A numeric vector of length 2 of figure width and height.
+#'   The length should be 2 (width, height). The unit is inch.
+#' @inheritParams r2rtf::rtf_page
+#' @inheritParams r2rtf::rtf_body
+#' @param footnotes A character vector of table footnotes.
+#' @param title Term "analysis", "observation" and "population") for collecting title
+#'   from metadata or a character vector of table titles.
+#' @param path_outdata A character string of the outdata path.
+#' @param path_outtable A character string of the outtable path.
+#'
+#' @return RTF file and source dataset for baseline characteristic table.
+#'
+#' @export
+#'
+#' @examples
+#' forestly_adsl$TRTA <- factor(
+#'   forestly_adsl$TRT01A,
+#'   levels = c("Xanomeline Low Dose", "Placebo"),
+#'   labels = c("Low Dose", "Placebo")
+#' )
+#' forestly_adae$TRTA <- factor(
+#'   forestly_adae$TRTA,
+#'   levels = c("Xanomeline Low Dose", "Placebo"),
+#'   labels = c("Low Dose", "Placebo")
+#' )
+#' outdata <- meta_forestly(
+#'   dataset_adsl = forestly_adsl,
+#'   dataset_adae = forestly_adae
+#' ) |>
+#'   prepare_ae_forestly()|>
+#'   format_ae_forestly()
+#'
+#' p1 <- substitute(
+#'   plot_dot(
+#'     tbl,
+#'     prop_cols = c("prop_1", "prop_2"),
+#'     y_var = "name",
+#'     label = c("Treatment", "Placebo")
+#'   )
+#' )
+#' p2 <- substitute(
+#'   plot_errorbar(
+#'     tbl,
+#'     ci_cols = c("diff_1", "lower_1", "upper_1"),
+#'     y_var = "name",
+#'     label = c("Treatment", "Placebo"),
+#'     legend_nrow = NULL,
+#'     theme = theme_panel(show_ticks = FALSE, show_text = FALSE)
+#'   )
+#' )
+#' p3 <- substitute(
+#'   table_panel(
+#'     tbl,
+#'     n_cols = c("n_1", "n_2"),
+#'     prop_cols = c("prop_1", "prop_2"),
+#'     y_var = "name",
+#'     theme = theme_panel(show_ticks = FALSE, show_text = FALSE),
+#'     x_label = c("Treatment \n n(%)", "Placebo \n n(%)")
+#'   )
+#' )
+#' outdata |> rtf_static_forestly(
+#'   plot_calls = c(p1, p2, p3),
+#'   source = "Source:  [CDISCpilot: adam-adsl; adae]",
+#'   path_outdata = tempfile(fileext = ".Rdata"),
+#'   path_outtable =  tempfile(fileext = ".rtf")
+#' )
+rtf_static_forestly <- function(
+    outdata,
+    plot_calls,
+    source,
+    parameter = "any", # only one parameter can be selected
+    n_rows = 25,
+    orientation = "portrait",
+    fig_size = c(6, 6),
+    title = c("analysis", "observation", "population"),
```

Comment:
this argument will confuse user who are not familiar with `metalite`. 

If we only allow limited combination of `title`, we may consider a `title_type` argument with value 1, 2, 3 to clarify what it can be achieved. 

If I am a user, I would infer `title` should allow me to customize the full title. if needed.

Reply:
Since argument `title` is also used in other `tlf` function, should we keep the same naming convention.

https://github.com/Merck/metalite.ae/blob/2e45d26325d7eec68aac8d1b140730b68c4dcf48/R/tlf_ae_specific.R#L63